### PR TITLE
GH-101097: Switch from standard interval notation to greater or less than signs for random.random()'s documentation

### LINKED ISSUE #101097: Typo in python random.random documentation
There is a typo in documentation [here](https://docs.python.org/3/library/random.html#random.random) regarding random.random()
`the next random floating point number in the range [0.0, 1.0).` 

It can be corrected by replacing ")" with "]"

### Linked PRs
* gh-101119
* gh-101246

### PR DESCRIPTION
**Pull request by andrewhong04**
Fixes: https://github.com/python/cpython/issues/101097

* Issue: gh-101097

### CODE REVIEW

File: Doc/library/random.rst [hunk 1]
```diff
@@ -294,7 +294,7 @@ be found in any statistics text.
 
 .. function:: random()
 
-   Return the next random floating point number in the range [0.0, 1.0).
+   Return the next random floating point number in the range ``0.0 <= x < 1.0``
```
Comment (rhettinger) [inline Doc/library/random.rst]: Please use a capital `X` which is the norm for random variables.

**Review by rhettinger — CHANGES_REQUESTED** [review]

**Comment by rhettinger** [comment]
Also update the corresponding docstring in `Lib/random.py`.

**Comment by novialriptide** [comment]
I have made the requested changes; please review again

File: Lib/random.py [hunk 2]
```diff
@@ -849,7 +849,7 @@ class SystemRandom(Random):
     """
 
     def random(self):
-        """Get the next random number in the range [0.0, 1.0)."""
+        """Return the next random floating point number in the range ``0.0 <= X < 1.0``"""
```
Comment (rhettinger) [inline Lib/random.py]: No backticks here.  This is a docstring and doesn't use markup.

Comment (rhettinger) [reply Lib/random.py]: Also, let's keep the wording as-is and only change the  interval.  I know it's a little different that the mains but that is okay.

File: Doc/library/random.rst [hunk 3]
```diff
@@ -21,8 +21,8 @@ lognormal, negative exponential, gamma, and beta distributions. For generating
 distributions of angles, the von Mises distribution is available.
 
 Almost all module functions depend on the basic function :func:`.random`, which
-generates a random float uniformly in the semi-open range [0.0, 1.0).  Python
-uses the Mersenne Twister as the core generator.  It produces 53-bit precision
+generates a random float uniformly in the semi-open range ``0.0 <= X < 1.0``.
```
Comment (rhettinger) [inline Doc/library/random.rst]: While we're at it, change "semi-open" to "half-open".

**Review by rhettinger — CHANGES_REQUESTED** [review]

**Comment by novialriptide** [comment]
I have made the requested changes; please review again

**Review by rhettinger — APPROVED** [review]

**Pull request merged**